From bd04dfb4d9dbf2d29c5a5c18bfce15f926996e5e Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Fri, 11 Sep 2020 18:26:47 +0200
Subject: [PATCH 1/2] Remove old kio_fonts hack in KCoreDirLister, hostname was
 stripped incorrectly

https://bugs.kde.org/show_bug.cgi?id=160057 worked around the fact that
kio_fonts was creating URLs like fonts://System/, and yet there's no
hostname during directory listing afterwards.

Nowadays:
1) this hack isn't needed anymore, kio_fonts does fonts:/System/
2) removing it actually fixes navigating from `dolphin fonts:/`
---
 src/core/kcoredirlister.cpp | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src/core/kcoredirlister.cpp b/src/core/kcoredirlister.cpp
index e9a756a4..8a548949 100644
--- a/src/core/kcoredirlister.cpp
+++ b/src/core/kcoredirlister.cpp
@@ -90,16 +90,6 @@ bool KCoreDirListerCache::listDir(KCoreDirLister *lister, const QUrl &_u,
     QUrl _url(_u);
     _url.setPath(QDir::cleanPath(_url.path())); // kill consecutive slashes
 
-    if (!_url.host().isEmpty() && KProtocolInfo::protocolClass(_url.scheme()) == QLatin1String(":local")
-            && _url.scheme() != QLatin1String("file")) {
-        // ":local" protocols ignore the hostname, so strip it out preventively - #160057
-        // kio_file is special cased since it does honor the hostname (by redirecting to e.g. smb)
-        _url.setHost(QString());
-        if (_keep == false) {
-            emit lister->redirection(_url);
-        }
-    }
-
     // like this we don't have to worry about trailing slashes any further
     _url = _url.adjusted(QUrl::StripTrailingSlash);
 
-- 
2.25.1

